Libraries/LibJS/AST.cpp

```diff
@@ -1088,8 +1088,7 @@ PropertyName MemberExpression::computed_property_name(Interpreter& interpreter)
     if (interpreter.exception())
         return {};
     ASSERT(!index.is_empty());
-    // FIXME: What about non-integer numbers tho.
-    if (index.is_number() && index.to_i32() >= 0)
+    if (index.to_number().is_finite_number() && (i32)index.to_double() == index.to_i32() && index.to_i32() >= 0)
```

Comment:
There's a lot of coercion hiding behind these calls. Perhaps we should coerce to number only once?